checkov/terraform/graph_builder/local_graph.py

```diff
@@ -172,90 +177,115 @@ def _build_edges(self) -> None:
+            ):
+                continue
+            referenced_vertices = get_referenced_vertices_in_value(
+                value=attribute_value,
+                aliases=aliases,
+                resources_types=resources_types,
+            )
+            for vertex_reference in referenced_vertices:
+                # for certain blocks such as data and resource, the block name is composed from several parts.
+                # the purpose of the loop is to avoid not finding the node if the name has several parts
+                sub_values = [remove_index_pattern_from_str(sub_value) for sub_value in vertex_reference.sub_parts]
+                for i in range(sub_values):
+                    reference_name = join_trimmed_strings(char_to_join=".", str_lst=sub_values, num_to_trim=i)
+                    if vertex.module_dependency:
+                        dest_node_index = self._find_vertex_index_relative_to_path(
+                            vertex_reference.block_type, reference_name, vertex.path, vertex.module_dependency,
+                            vertex.module_dependency_num
+                        )
+                        if dest_node_index == -1:
                             dest_node_index = self._find_vertex_index_relative_to_path(
-                                vertex_reference.block_type, reference_name, vertex.path, vertex.module_dependency, vertex.module_dependency_num
+                                vertex_reference.block_type, reference_name, vertex.path, vertex.path,
+                                vertex.module_dependency_num
                             )
-                            if dest_node_index == -1:
-                                dest_node_index = self._find_vertex_index_relative_to_path(
-                                    vertex_reference.block_type, reference_name, vertex.path, vertex.path, vertex.module_dependency_num
+                    else:
+                        dest_node_index = self._find_vertex_index_relative_to_path(
+                            vertex_reference.block_type, reference_name, vertex.path, vertex.module_dependency,
+                            vertex.module_dependency_num
+                        )
+                    if dest_node_index > -1 and origin_node_index > -1:
+                        if vertex_reference.block_type == BlockType.MODULE:
+                            try:
+                                self._connect_module(
+                                    sub_values, attribute_key, self.vertices[dest_node_index], origin_node_index,
+                                    cross_variable_edges
                                 )
+                            except Exception as e:
+                                logging.warning(
+                                    f"Module {self.vertices[dest_node_index]} does not have source attribute, skipping"
+                                )
+                                logging.warning(e, stack_info=True)
```

Comment:
```suggestion
                                logging.warning(
                                    f"Module {self.vertices[dest_node_index]} does not have source attribute, skipping", exc_info=True
                                )
```
I think this should be the same, no need to have two logging calls